data/datum.go

```diff
@@ -31,7 +31,7 @@ type Datum interface {
 	SetDeletedTime(deletedTime *time.Time)
 	SetDeletedUserID(deletedUserID *string)
 
-	UpdatesSummary() bool
+	UpdatesSummary(updatesSummary *map[string]bool)
```

Comment:
In this case just passing the map as is / no need for pointer and modifying it within the interface implementation, e.g., `updatesSummary["cgm"] = "foo"` should work fine.

Reply:
@lostlevels yep, good catch. 

@Roukoswarf The naming is also off - `UpdatesSummary` implies a boolean result. If this takes a map that's being modified the name should reflect that (e.g. `PopulateAffectedSummaryTypes(types map[string]struct{}) error`). If it simply returns the types which are affected it should probably be `GetAffectedSummaryTypes() []string, error` and let the caller deal with the logic. Since this is a base type, I prefer to avoid having summary specific logic and data structures.

Reply:
Inlined this logic into the API code as suggested by another review comment.